### ISSUE TITLE
`reportDuplicateImport` false negative when imports are aliased

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#89]
```py
from typing import cast as a, overload as a # no error
```
